### PR TITLE
[Project] Basic and Advanced Web Examples

### PR DESCRIPTION
**Pull request by crypblizz8**
Refactor the examples to be `basic` and `advanced`

Note: Vercel will probably break due to the pointing of examples

New Suggestion/ Build

### Basic
Dapps
- [x]  web3modal 
- [x]  (web3inbox)
- [x]  ethereum-provider 
- [x]  universal-provider-solana 

Wallets
- [x]  web3wallet

### Advanced
Dapps
- react-dapp-auth
- react-dapp-v2-cosmos-provider
- react-dapp-v2-ethers
- react-dapp-v2-web3js
- react-dapp-v2
- vue-dapp-auth

Wallets
- react-wallet-auth
- react-wallet-chat
- react-wallet-v2
- react-web3wallet

Settings
- [ ] Configure all vercels to build to the new directory

### CODE REVIEW

File: basic/dapps/universal-provider-solana/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    "dev": "vite",
+    "build": "tsc && vite build",
+    "lint": "eslint src --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
+    "preview": "vite preview"
+  },
+  "dependencies": {
+    "@esbuild-plugins/node-globals-polyfill": "^0.2.3",
+    "@solana/web3.js": "^1.78.4",
+    "@walletconnect/modal": "^2.6.1",
+    "@walletconnect/universal-provider": "2.10.0-canary-ac4c97",
```
Comment (crypblizz8) [inline basic/dapps/universal-provider-solana/package.json:16]: Can you elaborate the use of the canary here?

**Review by crypblizz8 — COMMENTED** [review]
LGTM so far. Have question about UProvider (Solana)

**Comment by crypblizz8** [comment]
Please talk to @bkrem on how best to test the builds with the `advanced` dapps and wallets as there is a Vercel Config part we need to do

File: basic/dapps/universal-provider-solana/package.json (hunk 1 above)
Comment (boidushya) [reply basic/dapps/universal-provider-solana/package.json:16]: good catch, updating it tonight, should be `2.10.0`

File: basic/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Basic Web Examples
+
```
Comment (bkrem) [inline basic/README.md:2]: Can we get a proper description here of what the intent is of these examples? E.g. to provide basic building blocks or templates...

File: basic/dapps/ethereum-provider/.codesandbox/tasks.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{
+  // These tasks will run in order when initializing your CodeSandbox project.
+  "setupTasks": [
+    {
+      "name": "Install Dependencies",
+      "command": "yarn install"
+    }
+  ],
+
+  // These tasks can be run from CodeSandbox. Running one will open a log in the app.
```
Comment (bkrem) [inline basic/dapps/ethereum-provider/.codesandbox/tasks.json:10]: Are these auto generated bc generally comments in JSON are invalid as shown here 🤔

File: basic/dapps/universal-provider-solana/package.json (hunk 1 above)
Comment (bkrem) [reply basic/dapps/universal-provider-solana/package.json:16]: Bump

File: basic/dapps/web3inbox/README.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Web3Inbox Example
+
+Head over to [GM Hackers Repository](https://github.com/WalletConnect/gm-hackers) for more information
```
Comment (bkrem) [inline basic/dapps/web3inbox/README.md:3]:
```suggestion
Head over to the [GM Hackers Repository](https://github.com/WalletConnect/gm-hackers) for more information.
```

**Review by bkrem — APPROVED** [review]

**Review by bkrem — APPROVED** [review]

File: basic/dapps/ethereum-provider/src/App.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import { ethers } from "ethers";
+import React, { useState } from "react";
+
+const projectId = import.meta.env.VITE_PROJECT_ID as string;
+
+// 1. Create a new EthereumProvider instance
+const provider = await EthereumProvider.init({
+  projectId,
+  chains: [1],
+  methods: ["eth_signTypedData", "eth_signTypedData_v4"],
```
Comment (ganchoradkov) [inline basic/dapps/ethereum-provider/src/App.tsx:13]:
```suggestion
  methods: ["personal_sign", "eth_sendTransaction"],
```
Given this is a basic example, should we use (especially as required) methods that are supported by every wallet? Imho, we should avoid any friction when devs attempt to use & connect using these examples

File: basic/dapps/ethereum-provider/.codesandbox/tasks.json (hunk 3 above)
Comment (boidushya) [reply basic/dapps/ethereum-provider/.codesandbox/tasks.json:10]: Yep autogenerated by codesandbox. Ran locally and seems to run properly w/o errors

**Review by ganchoradkov — APPROVED** [review]
awesome 💯

File: basic/dapps/universal-provider-solana/src/utils/helpers.ts [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    bs58.decode(address)
+  );
+}
+
+const isVersionedTransaction = (
+  transaction: Transaction | VersionedTransaction
+): transaction is VersionedTransaction => "version" in transaction;
+
+
+export const getProviderUrl = (chainId: string) => {
```
Comment (0xAsimetriq) [inline basic/dapps/universal-provider-solana/src/utils/helpers.ts:47]: Did we confirm that our rpc supports solana?

File: basic/dapps/web3modal/vite.config.ts.timestamp-1690122939308-5596491a6a4e3.mjs [hunk 7]
```diff
@@ -0,0 +1,10 @@
+// vite.config.ts
```
Comment (0xAsimetriq) [inline basic/dapps/web3modal/vite.config.ts.timestamp-1690122939308-5596491a6a4e3.mjs:1]: This looks autogenerated / like it shouldn't be included?

**Review by 0xAsimetriq — COMMENTED** [review]
Left few comments, also personal preference, but would be good to base examples on `npm` instead of yarn (I think this uses v1 yarn which is deprecated / not updated for a year)

File: basic/dapps/universal-provider-solana/src/utils/helpers.ts (hunk 6 above)
Comment (boidushya) [reply basic/dapps/universal-provider-solana/src/utils/helpers.ts:47]: Yep, its [supported](https://github.com/WalletConnect/blockchain-api/blob/master/SUPPORTED_CHAINS.md#http-rpc) 🖖

File: basic/dapps/universal-provider-solana/src/utils/helpers.ts [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+    );
+
+    return {
+      method: "solana_signMessage",
+      address,
+      valid,
+      result: result.signature,
+    };
+    //eslint-disable-next-line
+  } catch (error: any) {
```
[the remark this answers is not in the document]
Comment (boidushya) [reply basic/dapps/universal-provider-solana/src/utils/helpers.ts:84]: image
Nope, better to keep as is for the time being imho

**Comment by boidushya** [comment]
> Left few comments, also personal preference, but would be good to base examples on `npm` instead of yarn (I think this uses v1 yarn which is deprecated / not updated for a year)

SGTM, lets do this later after we get this merged

**Review by boidushya — APPROVED** [review]

**Pull request merged**